drivers/serial/serial.c

```diff
@@ -1761,7 +1771,7 @@ void uart_connected(FAR uart_dev_t *dev, bool connected)
 void uart_reset_sem(FAR uart_dev_t *dev)
 {
   nxsem_reset(&dev->xmitsem,  0);
```

Comment:
should we change 0 to dev->xmitwaiting?

Reply:
The same thing. No. It is checked in the datasend interrupt. After closing the driver no interrupt will happen.